apps/publications/src/app/pages/create-publication-page/create-single-publication-page/create-single-publication-page.component.ts

```diff
@@ -0,0 +1,177 @@
+import { StepperSelectionEvent } from '@angular/cdk/stepper';
+import { MatTabGroup } from '@angular/material/tabs';
+
+const moment =  _moment;
+
+export const YEAR_MODE_FORMATS = {
+  parse: {
+    dateInput: 'YYYY',
+  },
+  display: {
+    dateInput: 'YYYY',
+    monthYearLabel: 'MMM YYYY',
+    dateA11yLabel: 'LL',
+    monthYearA11yLabel: 'MMMM YYYY',
+  },
+};
+
+@Component({
+  selector: 'perun-web-apps-create-single-publication-page',
+  templateUrl: './create-single-publication-page.component.html',
+  styleUrls: ['./create-single-publication-page.component.scss'],
+  providers: [
+    {
+      provide: DateAdapter,
+      useClass: MomentDateAdapter,
+      deps: [MAT_DATE_LOCALE],
+    },
+    {provide: MAT_DATE_FORMATS, useValue: YEAR_MODE_FORMATS}
+  ]
+})
+export class CreateSinglePublicationPageComponent implements OnInit {
+
+  constructor(private formBuilder: FormBuilder,
+              private cabinetService: CabinetManagerService,
+              private dialog: MatDialog,
+              private router: Router,
+              private notificator: NotificatorService,
+              private translate: TranslateService,) { }
+
+  publicationControl: FormGroup;
+  similarPublications: PublicationForGUI[] = [];
+  filteredPublications: PublicationForGUI[] =[];
+  categories: Category[] = [];
+
+  publication: PublicationForGUI = null;
+
+  authorsSelection: SelectionModel<Author> = new SelectionModel<Author>(true, []);
+  thanksSelection: SelectionModel<ThanksForGUI> = new SelectionModel<ThanksForGUI>(true, []);
+
+  innerLoading = false;
+
+  maxYear: Moment;
+
+  loading = false;
+  duplicateCheck = false;
+
+  selectedPubId = null;
+  selectedPubTitle = '';
+
+  ngOnInit(): void {
+    this.loading = true;
+    this.publicationControl = this.formBuilder.group({
+      title: ['', Validators.required],
+      year: ['', Validators.required],
+      category: ['', Validators.required],
+      ISBN: [''],
+      DOI: [''],
+      cite: ['', Validators.required],
+      }
+    );
+
+    this.maxYear = moment();
+
+    this.cabinetService.getCategories().subscribe(categories => {
+      this.categories = categories;
+        this.loading = false;
+    });
+  }
+
+  chosenYearHandler(normalizedYear: Moment, datepicker: any) {
+    this.publicationControl.get('year').setValue(normalizedYear);
+    datepicker.close();
+  }
+
+  createPublication(){
+    this.innerLoading = true;
+    const publicationInput: any = {
+      publication: {
+        title: this.publicationControl.get("title").value,
+        categoryId: this.publicationControl.get("category").value.id,
+        year: this.publicationControl.get("year").value.year,
+        isbn: this.publicationControl.get("ISBN").value,
+        doi: this.publicationControl.get("DOI").value,
+        main: this.publicationControl.get("cite").value,
+      }
+    };
+    this.cabinetService.createPublication(publicationInput).subscribe(publication => {
```

Comment:
please add to input property "createdBy" with the email of the user who is creating the publication